tests/test_history.py

```diff
@@ -1592,6 +1592,49 @@ def test_daily_history_blended_gaps(self, field):
                                            err_msg='field={0} minute={1}'.
                                            format(field, minute))
 
+    @parameterized.expand([(("bar_count%s" % x), x) for x in [1, 2, 3]])
+    def test_daily_history_minute_gaps_price_ffill(self, test_name, bar_count):
+        # Make sure we use the previous day's value when there's been no volume
+        # yet today.
+
+        # January 5 2015 is the first day, and there is volume only every
+        # 10 minutes.
+        for day_idx, day in enumerate([pd.Timestamp('2015-01-05', tz='UTC'),
+                                       pd.Timestamp('2015-01-06', tz='UTC')]):
+
+            minutes = self.trading_calendar.minutes_for_session(day)
+
+            equity_cal = self.trading_calendars[Equity]
+            equity_minutes = equity_cal.minutes_for_session(day)
+            equity_close = equity_minutes[-1]
+
+            # minute data, baseline:
+            # Jan 5: 2 to 391
+            # Jan 6: 392 to 781
+            # Jan 7: 782 to 1171
+            for minute in minutes:
+                idx = equity_minutes.searchsorted(min(minute, equity_close))
+
+                window = self.data_portal.get_history_window(
+                    [self.ASSET3],
+                    minute,
+                    bar_count,
+                    '1d',
+                    'price',
+                    'minute',
+                )[self.ASSET3]
+
+                self.assertEqual(len(window), bar_count)
+
+                if not day_idx and idx + 1 < 10:
+                    self.assertTrue(np.isnan(window[-1]))
```

Comment:
I wouldn't block on it, but I would recommend picking a few specific times and expected values, crafted to test each condition we expect. That would 1) speed up the test time 2) make the narrative of what is being tested and why more clear.

Reply:
Seconded 👍

Reply:
Good idea! Updated.

Reply:
Thanks. In most cases I'd recommend the specific Timestamps, i.e. hardcoding `pd.Timestamp('2017-01-09 10:44'`); however since this test is iterating over multiple days, I agree with using the index to select the time.

Reply:
Yea, in this case, I saw the important information as being the open or close or the temporal relationship to the open, etc, as opposed to exact times, so I represented them that way.